server/src/main/java/org/elasticsearch/index/snapshots/blobstore/SlicedInputStream.java

```diff
@@ -99,4 +110,38 @@ public final int available() throws IOException {
         return stream == null ? 0 : stream.available();
     }
 
+    @Override
+    public boolean markSupported() {
+        return true;
+    }
+
+    @Override
+    public void mark(int readLimit) {
```

Comment:
Both `mark` and `reset` are defined as syncrhonized on `InputStream`. This class seems to assume single threaded usage so no need for `synchronized`. But think I'd ask just to be explicit.

Reply:
Yes this class is single threaded.

I searched a bit more around the `synchronized` on the base class mark/reset and found [this JDK bug](https://bugs.openjdk.org/browse/JDK-8292631) and [this](https://stackoverflow.com/a/30775991), so it was an oversight in the jdk code and should be removed in recent jdk versions.